server/routes/api/user.js

```diff
@@ -1,21 +1,56 @@
 const express = require('express');
 const router = express.Router();
 
-/* GET users listing. */
-router.get('/', (req, res, next) => {
-  res.send(
-    'This URL will handle POST, PUT, and DELETE requests relating to user accounts.'
-  );
-});
+/* PATCH the user settings 
+req format:
+{
+  model-value: new-value
+}
+Locked values:
+* passwordHash
+*/
+router.patch('/', async (req, res, next) => {
+  let changed = false;
+  for (let key of Object.keys(req.body)) {
+    if (
+      typeof req.user.get(key) !== 'undefined' &&
+      key != 'passwordHash' &&
+      key != 'id' &&
+      key != 'role' &&
+      key != 'balance'
+    ) {
+      // TODO conver req.user to a db object
+      req.user.set(key, req.body[key]);
+      changed = true;
+    }
+  }
+
+  if (changed) {
+    req.user.save();
+    res.status(204);
+  } else {
+    res.status(400);
+  }
 
-/* POST a new User. */
-router.post('/', (req, res, next) => {
-  res.send('This endpoint will try to create a new user.');
+  res.send();
 });
 
-/* PUT a new User. */
-router.put('/', (req, res, next) => {
-  res.send('This endpoint will update the users information.');
+router.get('/', async (req, res, next) => {
+  const ret = {};
+  for (let key of Object.keys(req.user.dataValues)) {
+    if (
+      typeof req.user.get(key) !== 'undefined' &&
+      key != 'passwordHash' &&
+      key != 'id'
+    ) {
+      ret[key] = req.user.get(key);
+    }
+  }
+
+  res.status(200);
+
+  res.type('application/json');
+  res.send(JSON.stringify(ret));
```

Comment:
Here's a few tips to make things simpler. A nice feature of express and most other node-based HTTP frameworks is that you can just pass in any JavaScript object's `.send()` method (or equivalent) and it will automatically JSON stringify it and set the content type header to `application/json`, saving a bit of repetition.

Also, with express, the response object implements the builder pattern, so you can chain together calls to make it more concise. With those two tricks, you can easily simplify the following to this one line:
```suggestion
res.status(200).send(ret);
```

Putting the status code on the same line as send is super common practice in express. However, with res.send(), you don't have to specify status if you're just sending a 200. If you don't specify a status and no errors are thrown, a 200 is set by default, so you could even do as simple as:
```suggestion
  res.send(ret);
```